model/map/src/main/java/org/keycloak/models/map/common/ExpirableEntity.java

```diff
@@ -44,4 +44,18 @@ public interface ExpirableEntity extends AbstractEntity {
      * @param expiration a timestamp in milliseconds since The Epoch or {@code null} if this entity never expires.
      */
     void setExpiration(Long expiration);
+
+    /**
+     * Returns a point in time (timestamp in milliseconds since The Epoch) when this entity started its existence
+     *
+     * @return a timestamp in milliseconds since The Epoch or {@code null} when the time is unknown
+     */
+    Long getTimestamp();
```

Comment:
Looking at the docs of RootAuthenticationSessionModel#getTimestamp(), I read "Returns timestamp when the root authentication session was created or updated". 

Without so much knowledge about the code at hand, I wonder if this JavaDoc should be updated, and how the current code behaves with regards to updating this field. Looking at the code it seems that it expiry is always updated when the timestamp is updated as well.

Will an updated expiry time also be propagated to HotRod? Looking at the code I didn't find where the initial expiry or an update expiry is passed to HotRod. 

(applies also to setTimestamp())

Reply:
Not sure what you mean by the first part of the comment, could you rephrase, please?

For the second part, HotRod currently doesn't use expiration in any way to set cache entry lifespan, this PR is just preparing storage to make this simply possible. This will be done in this follow-up task: https://github.com/keycloak/keycloak/issues/12068

Reply:
Reading that HotRod doesn't use the expiration as of now, I rephrase it as: Please update the JavaDoc from 

"Returns a point in time (timestamp in milliseconds since The Epoch) when this entity started its existence" 

to 

"Returns a point in time (timestamp in milliseconds since The Epoch) when the entity was created or updated."

as the field can be updated to indicate for example a refreshed session, like this is stated in the JavaDoc of RootAuthenticationSessionModel#getTimestamp.

Reply:
Hmm, I see it now thanks. That makes sense, I will do this.